.maintain/common/lib.sh

```diff
@@ -115,3 +106,51 @@ has_runtime_changes() {
     return 1
   fi
 }
+
+# Add a label to a PR in a given repo. Doesn't error if label already present
+# repo: paritytech/substrate
+# label: A7-needspolkadotpr
+# PR: 1234
+add_label() {
+  repo=$1
+  pr=$2
+  label=$3
+  curl -X POST \
+    --data "[\"$label\"]" \
+    -H "Accept: application/vnd.github.v3+json" \
+    -H "Authorization: token $(github_token)" \
+    -s "$api_base/$repo/issues/$pr/labels"
+}
+
+# Remove a label from a PR in a given repo. Doesn't error if label already absent
+# repo: paritytech/substrate
+# label: A7-needspolkadotpr
+# PR: 1234
+remove_label() {
+  repo=$1
+  pr=$2
+  # Escape the labels... we use emojis quite often, so this is required
+  # to not break things.
+  # Source below: (with od instead of xxd since it's in IEEE Std 1003.1-2008)
+  # https://stackoverflow.com/questions/12735450/delete-using-curl-with-encoded-url
+  label=$(printf '%s' "$3" | od -A n -w1000 -t x1 | tr -d ' ' | sed 's/\(..\)/%\1/g')
+
+  curl -X DELETE \
+    -H "Accept: application/vnd.github.v3+json" \
+    -H "Authorization: token $(github_token)" \
+    -s "$api_base/$repo/issues/$pr/labels/$label"
+}
+
+# Useful shim for getting the right GITHUB_TOKEN, depending on whether we're
+# in Github or Gitlab CI
+github_token() {
+  # These will exist if the function is called in Gitlab.
+  # If the function's called in Github, we should have GITHUB_ACCESS_TOKEN set
```

Comment:
will this token then be used by the curl calls? otherwise setting it from another else block would help.